astroid/brain/brain_namedtuple_enum.py

```diff
@@ -413,6 +413,8 @@ def infer_enum_class(node: nodes.ClassDef) -> nodes.ClassDef:
                 else:
                     targets = stmt.targets
             elif isinstance(stmt, nodes.AnnAssign):
+                if all(s is astroid.Uninferable for s in stmt.assigned_stmts()):
```

Comment:
Checking Uninferable might catch other truly uninferable assignment values, maybe?

```suggestion
                if stmt.value is None:
```

Reply:
This is the issue - we need to allow `red: int = "something"` but prevent `red: int` from entering the `__members__` container. Using `assigned_stmts()` let's me distinguish these two scenarios since `red: int` will have a value of `None` in the `stmt.value` approach.

Reply:
Ah, if you could flesh out just a bit further that would help me understand. 👍 

> since red: int will have a value of None in the stmt.value approach

Isn't this exactly what we want? Tests pass with my suggestion and continue to distinguish the two cases.

Reply:
Checking for a None value is the convention (I think) for elsewhere in pylint/astroid for checking annotations without values.

Reply:
Let me try it out! (I thought that I had already tried this during development and it wasn't working but perhaps it was something a bit different)

Reply:
Needed to move this logic further down the function so that something would be returned by the function - this manifested itself in a failing unittest testing an enum class with a single (annotated with no value) member.

Reply:
Funny enough the unittest only failed on 3.12